File: NCore/Commands/Commandtogglespectator.cs
using NCore;
using System.Threading.Tasks;

namespace NCore
{
    public class Commandtogglespectator : Command
    {
        public Commandtogglespectator() : base("togglespectator", "Переключает режим наблюдателя.", "togglespectator")
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            if (!sender.GetAdmin())
            {
                await sender.SendMessage("У Вас недостаточно прав!");
                return true;
            }

            if (!sender.IsPlayer)
            {
                return false;
            }

            if (args.Length == 0)
            {
                NetworkPlayer p = (NetworkPlayer)sender;
                if (p.IsSpectator)
                {
                    await p.Survival();
                }
                else
                {
                    await p.Spectator();
                }

                return true;
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandtogglespectator.vb
Imports NetcraftServer

Public Class Commandtogglespectator
    Inherits Command
    Sub New()
        MyBase.New("togglespectator", "Переключает режим наблюдателя.", "togglespectator")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If Not sender.IsPlayer Then
            Return False
        End If
        If args.Length = 0 Then
            Dim p As NetworkPlayer = sender
            If p.IsSpectator Then
                p.Survival()
            Else
                p.Spectator()
            End If
            Return True
        End If
        Return False
    End Function
End Class
